recipes/natural_language_processing/summarizer/app/summarizer.py

```diff
@@ -49,12 +50,12 @@ def read_file(file):
     file_type = file.type
     
     if file_type == "application/pdf":
-        with open(file.name, "wb") as f:
+        temp = tempfile.NamedTemporaryFile()
```

Comment:
You still need to remove this when you are done with it?

Reply:
When I tested it the tempfile deleted after the file was no longer being used.

Reply:
Is this some magic in Python?

Reply:
```
Use the [tempfile module](http://docs.python.org/2/library/tempfile.html); it creates temporary files that auto-delete.

From the [tempfile.NamedTemporaryFile() documentation](http://docs.python.org/2/library/tempfile.html#tempfile.NamedTemporaryFile):

    If delete is true (the default), the file is deleted as soon as it is closed.
```
Cool